core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -64,42 +53,98 @@ trait EntityComponent {
 
   import driver.api._
 
-  class EntityTable(tag: Tag) extends Table[EntityRecord](tag, "ENTITY") {
+  sealed abstract class EntityTableBase[RECORD_TYPE <: EntityRecordBase](tag: Tag) extends Table[RECORD_TYPE](tag, "ENTITY") {
     def id = column[Long]("id", O.PrimaryKey, O.AutoInc)
     def name = column[String]("name", O.Length(254))
     def entityType = column[String]("entity_type", O.Length(254))
     def workspaceId = column[UUID]("workspace_id")
     def version = column[Long]("record_version")
-    def allAttributeValues = column[Option[String]]("all_attribute_values")
     def deleted = column[Boolean]("deleted")
     def deletedDate = column[Option[Timestamp]]("deleted_date")
 
     def workspace = foreignKey("FK_ENTITY_WORKSPACE", workspaceId, workspaceQuery)(_.id)
     def uniqueTypeName = index("idx_entity_type_name", (workspaceId, entityType, name), unique = true)
+  }
 
-    def * = (id, name, entityType, workspaceId, version, allAttributeValues, deleted, deletedDate) <> (EntityRecord.tupled, EntityRecord.unapply)
+  class EntityTable(tag: Tag) extends EntityTableBase[EntityRecord](tag) {
+    def * = (id, name, entityType, workspaceId, version, deleted, deletedDate) <> (EntityRecord.tupled, EntityRecord.unapply)
+  }
+
+  class EntityTableWithInlineAttributes(tag: Tag) extends EntityTableBase[EntityRecordWithInlineAttributes](tag) {
+    def allAttributeValues = column[Option[String]]("all_attribute_values")
+
+    def * = (id, name, entityType, workspaceId, version, allAttributeValues, deleted, deletedDate) <> (EntityRecordWithInlineAttributes.tupled, EntityRecordWithInlineAttributes.unapply)
+  }
+
+  object entityQueryWithInlineAttributes extends TableQuery(new EntityTableWithInlineAttributes(_)) {
+    type EntityQueryWithInlineAttributes = Query[EntityTableWithInlineAttributes, EntityRecordWithInlineAttributes, Seq]
+
+    // only used by tests
+    def findEntityByName(workspaceId: UUID, entityType: String, entityName: String): EntityQueryWithInlineAttributes = {
+      filter(entRec => entRec.name === entityName && entRec.entityType === entityType && entRec.workspaceId === workspaceId)
+    }
+
+    private def createAllAttributesString(entity: Entity): Option[String] = {
+      val trimToSize = EntityComponent.allAttributeValuesColumnSize
+      Option(s"${entity.name} ${entity.attributes.values.filterNot(_.isInstanceOf[AttributeList[_]]).map(AttributeStringifier(_)).mkString(" ")}".toLowerCase.take(trimToSize))
+    }
```

Comment:
This should really recurse over `entity.attributes.values` until it reaches `trimToSize `, instead of potentially creating a giant string and `take`-ing the first `trimToSize` chars of it. I don't know if we run into this ever in production but it could be a cause of memory issues.

This PR does no harm and maintains the current behavior. I'm fine if a recursion optimization comes in a future PR.

Reply:
added